nature/models.py

```diff
@@ -234,44 +204,48 @@ class Meta:
         db_table = 'kohde_historia'
 
 
-class FeaturePublication(NatureModel):
-    feature_id = models.ForeignKey(Feature, models.CASCADE, db_column='kohdeid')
-    publication_id = models.ForeignKey(Publication, models.CASCADE, db_column='julkid')
+class FeaturePublication(models.Model):
+    """Through model for Feature & Publication m2m relation"""
+    feature = models.ForeignKey(Feature, models.CASCADE, db_column='kohdeid')
+    publication = models.ForeignKey(Publication, models.CASCADE, db_column='julkid')
 
     class Meta:
         managed = False
         db_table = 'kohde_julk'
-        unique_together = (('feature_id', 'publication_id'),)
+        unique_together = (('feature', 'publication'),)
 
 
-class FeatureLink(ProtectedNatureModel):
-    id = models.IntegerField(primary_key=True)
+class FeatureLink(models.Model):
     feature = models.ForeignKey(Feature, models.CASCADE, db_column='tekstiid', related_name='links')
     link = models.CharField(max_length=4000, blank=True, null=True, db_column='linkki')
     text = models.CharField(max_length=4000, blank=True, null=True, db_column='linkkiteksti')
-    type = models.ForeignKey('LinkType', models.PROTECT, db_column='tyyppiid')
+    link_type = models.ForeignKey('LinkType', models.PROTECT, db_column='tyyppiid')
     ordering = models.IntegerField(blank=True, null=True, db_column='jarjestys')
     link_text = models.CharField(max_length=1000, blank=True, null=True, db_column='linkin_teksti')
+    protection_level = models.ForeignKey('ProtectionLevel', models.PROTECT, db_column='suojaustasoid')
 
     class Meta:
         managed = False
         ordering = ['id']
         db_table = 'kohdelinkki'
 
+    def __str__(self):
+        return self.link
+
 
-class SpeciesRegulation(NatureModel):
-    species_id = models.ForeignKey('Species', models.CASCADE, db_column='lajid')
-    regulation_id = models.ForeignKey('Regulation', models.CASCADE, db_column='saaid')
+class SpeciesRegulation(models.Model):
+    """Through model for Species & Regulation m2m relation"""
+    species = models.ForeignKey('Species', models.CASCADE, db_column='lajid')
+    regulation = models.ForeignKey('Regulation', models.CASCADE, db_column='saaid')
 
     class Meta:
         managed = False
         db_table = 'laj_saa'
-        unique_together = (('species_id', 'regulation_id'),)
+        unique_together = (('species', 'regulation'),)
 
 
-class Observation(ProtectedNatureModel):
-    id = models.IntegerField(primary_key=True)
-    location = models.ForeignKey(Feature, models.PROTECT, db_column='kohdeid', related_name='observations')
```

Comment:
I guess this should be `feature`?

Reply:
Most likely that would be a better name :) I tried to be descriptive here, so that the REST user understands what meaning the Feature has for the Observation, i.e. this is the location that the observation was made.